### ISSUE TITLE
False positive: Incorrect error when ENTRYPOINT args contain commas

### ISSUE THREAD

**Issue by dikarel** [RedCoolBeans/dockerlint#19]
### Issue details

If any of the ENTRYPOINT args contain a comma, it will be marked as an error even though it's inside double-quotes.

### Steps to reproduce/test case

1. This valid input
  ```
  # Dockerfile
  FROM alpine
  ENTRYPOINT ["echo", "hello,world"]
  ```

2. Gives off this incorrect warning
  ![screen shot 2016-11-01 at 6 28 28 pm](https://cloud.githubusercontent.com/assets/15131271/19888404/08e60a72-a061-11e6-84ff-d5f5dafa32d8.png)

### Affected version(s)

dockerlint version 0.2.0, on OSX

**Comment by dikarel** [comment]
I looked at the `checks.coffee`, and it appears the linter checks for compliance using the following strategy:

1. Split into components using comma as the delimiter
2. Check each component with regex

It appears to me that it'd be more straightforward and correct(tm) to check for compliance via JSON.parse, as opposed to trying to do it via regex. This is in fact how the official [Dockerfile parser](https://github.com/docker/docker/blob/b248de7e332b6e67b08a8981f68060e6ae629ccf/builder/dockerfile/parser/line_parsers.go#L299) does it*. Thoughts?

---

\* First, try parsing it as JSON. If it can't be parsed into a string array, then assume the entire thing is a single string

**Comment by till** [comment]
With dockerlint@0.2.0 I found another bug:

This is an error:

```
FROM alpine:3.5
ENTRYPOINT [ "node", "server.js" ]
```

(Notice the whitespace around it?)

This is OK:

```
FROM alpine:3.5
ENTRYPOINT ["node", "server.js"]
```

**Comment by till** [comment]
Here's another one:

````
ENV app_root /foobar
WORKDIR ${app_root}
```

Complains that it's not an absolute path.

**Comment by rnagy** [comment]
This issue has been fixed by commit c0a9975032d2be09625873948e40469a5e6fb8b4

**Comment by till** [comment]
@rnagy should I create a new issue?

```
$ node bin/dockerlint.js -f ./test/Dockerfile-entrypoint 
ERROR: Arguments to ENTRYPOINT in exec form must not contain single quotes on line 2

ERROR: ./test/Dockerfile-entrypoint failed.
$ cat ./test/Dockerfile-entrypoint
FROM alpine:3.5
ENTRYPOINT [ "node", "server.js" ]

**Comment by rnagy** [comment]
This is a different issue which I just fixed in commit 2f60ffa6ad8fce3f816dde586ad6c55d10ada5f6
